src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,234 @@
+// featbin/wav-reverberate.cc
+
+// Copyright 2015  Tom Ko
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "feat/wave-reader.h"
+#include "feat/signal.h"
+
+namespace kaldi {
+
+void AddVectorsOfUnequalLength(const Vector<BaseFloat> &filter, Vector<BaseFloat> *signal) {
+  for (int po = 0; po < signal->Dim(); po += filter.Dim()) {
+    int block_length = filter.Dim();
+    if (signal->Dim() - po < block_length) block_length = signal->Dim() - po;
+    signal->Range(po, block_length).AddVec(1.0, filter.Range(0, block_length));
+  }
+}
+
+BaseFloat MaxAbsolute(const Vector<BaseFloat> &vector) {
+  return std::max(std::abs(vector.Max()), std::abs(vector.Min()));
+}
+
+BaseFloat ComputeEarlyReverbEnergy(const Vector<BaseFloat> &rir, const Vector<BaseFloat> &signal,
+                                   BaseFloat samp_freq) {
+  int peak_index = 0;
+  rir.Max(&peak_index);
+  KALDI_VLOG(1) << "peak index is " << peak_index;
+
```

Comment:
Add a brief text comment here describing what you are doing.

"Early reverberation component of the signal is composed of reflections within 0.05 seconds of the direct path signal (assumed to be the peak of the room impulse response). This function computes the energy in this early reverberation component of the signal."